src/movie/movie.controller.ts

```diff
@@ -0,0 +1,56 @@
+import { Controller, Get, Post, Body, Patch, Param, Delete, Query } from '@nestjs/common';
+import { MovieService } from './movie.service';
+import { CreateMovieDto } from './dto/create-movie.dto';
+import { UpdateMovieDto } from './dto/update-movie.dto';
+
+@Controller('movie')
+export class MovieController {
+  constructor(private readonly movieService: MovieService) {}
+
+  @Post()
+  create(@Body() createMovieDto: CreateMovieDto) {
+    return this.movieService.create(createMovieDto);
+  }
+
+  @Get()
+  findAll(
+    @Query('filter') filter: string, 
+    @Query('page') page: number = 1, 
+    @Query('limit') limit: number = 10
+  ) {
+    return this.movieService.findAll(filter, page, limit);
```

Comment:
You shouldn't return your database entity to the client. Always create a DTO and map the entity to the DTO before returning it to the client.
This helps for hiding attributes that you might not want to expose and also allows more flexibility in how objects are transferred to the client (We don't always want the client to receive data the same way it's stored in the database)